### PR TITLE
fix: update user id to cache and validate channel

### PR DESCRIPTION
**Pull request by im6h**
What purpose of the changes?
- add validate channel reply
- add cache time to user id

### CODE REVIEW

File: logic/autorename.go [hunk 1]
```diff
@@ -1,77 +1,97 @@
...
+		log.Info().Msgf("Rename from %s has processed", r.Author.Username)
+		return
+	}
 
-    msg := internal.Template{
-        Message: fmt.Sprintf("<@%s> đổi tên <@%s> thành %s", config.Bots[0], nick.oldNick, nick.newNick),
-    }
+	cacheNickName.Set(nick.userId, 1)
+	msg := internal.Template{
+		Message: fmt.Sprintf("<@%s> đổi tên <@%s> thành %s", config.Bots[0], nick.userId, nick.newNick),
```
Comment (nctam) [inline logic/autorename.go:75]: let's use our bot instead

File: logic/autorename.go [hunk 2]
```diff
@@ -1,77 +1,97 @@
...
+
+var (
+	cacheNickName cache.ExpiringCache
+	_             AutoRename = &GeneralAutoRename{}
 )
 
 func init() {
-    config = server.AppConfig
+	config = server.AppConfig
+	cacheNickName = cache.NewTTL(5*cacheTTL*time.Second, 5*cacheTTL*time.Second)
```
Comment (nctam) [inline logic/autorename.go]: let's move `5` to the config file, let's try *Minute instead

**Review by nctam — APPROVED** [review]

File: logic/autorename.go (hunk 1 above)
Comment (im6h) [reply logic/autorename.go:75]: I implement your suggestion in the next PR:

https://github.com/nctam/frog-chan/pull/9

**Comment by im6h** [comment]
Please check the PR if you have free time. Thanks for your comment

**Pull request merged**